src/routers/TabRouter.js

```diff
@@ -105,8 +105,12 @@ export default (
       const activeTabLastState = state.routes[state.index];
       const activeTabRouter = tabRouters[order[state.index]];
       if (activeTabRouter) {
+        const actionForChild =
```

Comment:
Wait, I'm confused... why is `action.action` *ever* passed in here?

The only case where it's relevant is `NavigationActions.NAVIGATE`, and in that case it's handled separately (see `if (action.action) {` on line 157). I think the point of this block here is for allowing the current child router to handle actions that are targeted to it, like a simple `{ type: NavigationActions.NAVIGATE,  routeName: "anotherChildRoute" }`. In cases where the action is targeted at the parent router, I think the expected result should be that `activeTabState === activeTabLastState`, and the logic below in `NavigationActions.NAVIGATE` handles the action. By passing the sub-action in here we circumvent that logic entirely.

As far as I understand, the right solution should be to never use `action.action` here, but I'm not as familiar with the code here as I'd like to be. Curious to here others' opinions.

Reply:
Interesting... yeah, that makes sense to me, but if I implement it I get another test failing. Not completely sure why - will investigate further